spock-core/src/main/java/org/spockframework/runtime/model/SpecInfo.java

```diff
@@ -168,6 +169,10 @@ public MethodInfo getInitializerMethod() {
     return initializerMethod;
   }
 
+  public Iterable<MethodInfo> getAllInitializerMethods() {
```

Comment:
WDYT about Iterable vs List as return types here? getFixtureMethods is the only one that is currently returning only an iterable, everything else returns a List and getAllFeatures is returning a List already.

Reply:
I don't have strong feelings about it (or any).
I just created it consistent with `getAllFixtureMethods`.
I have no idea why that and `getFixtureMethods` return `Iterable` while all others return `List`.
Maybe it was do differentiate between ordered and non-ordered?
But actually they also look ordered.
So yeah, we can also make all return `List`, those two also and documenting a potentially breaking change?
Do you want to do it or do you want me to change it?

Reply:
Is it a potentially breaking change? `List` implements `Iterable`, so it is a covariant change and should be compatible to all current users. The only thing that could break would be if someone used reflection to search for the method via its return type.

Reply:
It is a source-compatible change, but not a binary compatible change.
The return type is part of the signature when doing virtual method resolution.
So if you have for example a plugin compiled against the old state and try to execute it against the new state without recompiling, the execution should fail for not finding the method to execute.
Or do I have something wrong in mind?

Reply:
Yep:
```plain
java.lang.NoSuchMethodError: 'java.lang.Iterable org.spockframework.runtime.model.SpecInfo.getAllFixtureMethods()'
```

And it was iterable because it used `CollectionUtil.concat` which returns an `Iterable` and does not iterate the iterables upfront but lazily when needed.

I changed all to `List` now and added a breaking changes note but did not add it to the migration guide.

If with the new information you think it should stay `Iterable` or all should be `Iterable` just tell me and I change it again.